cluster/src/main/java/org/apache/iotdb/cluster/query/reader/ClusterReaderFactory.java

```diff
@@ -507,11 +541,16 @@ private SeriesReader getSeriesReader(
       Filter valueFilter,
       QueryContext context,
       Node header,
-      boolean ascending)
+      int raftId,
+      boolean ascending,
+      Set<Integer> requiredSlots)
       throws StorageEngineException, QueryProcessException {
     ClusterQueryUtils.checkPathExistence(path);
-    List<Integer> nodeSlots =
-        ((SlotPartitionTable) metaGroupMember.getPartitionTable()).getNodeSlots(header);
+    if (requiredSlots == null) {
```

Comment:
Please add some comments for this if statement.
BTW, why don't judge this in other functions such as `getMultSeriesReader`?

Reply:
I have added comments.
`why don't judge this in other functions such as getMultSeriesReader?`
Because all other readers will call this function eventually.